manifests/network/dns.pp

```diff
@@ -12,9 +11,9 @@
     $domains     = $_config['domains']
     $nameservers = $_config['nameservers']
     $options     = $_config['options']
-    validate_array($domains)
-    validate_array($nameservers)
-    validate_array($options)
+    assert_type(Array, $domains)
```

Comment:
There is a preexisting bug in this code.  See https://github.com/voxpupuli/puppet-system/pull/38#issuecomment-345016236